locator/token_metadata.cc

```diff
@@ -1146,6 +1146,20 @@ token_metadata::set_version_tracker(version_tracker_t tracker) {
     _impl->set_version_tracker(std::move(tracker));
 }
 
+version_tracker::~version_tracker() {
+    if (_stalled) {
+        auto d = std::chrono::duration_cast<std::chrono::duration<float>>(
+                std::chrono::steady_clock::now() - *_stalled);
+        tlogger.warn("topology version {} held for {:.3f} [s], released at: {}", _version, d.count(), seastar::current_backtrace());
+    }
+}
```

Comment:
So as far as I understand this will be called only when last reference to shared_token_metadata is destroyed, not for every holder, right?

Reply:
Right, only last ref, but to token_metadata (not shared_token_metadata). version_tracker is destroyed with token_metadata object.